### ISSUE TITLE
Allow inheriting from a base page without requiring `base.BuildRenderTree(__builder)`

### ISSUE THREAD

**Issue by Eagle3386** [dotnet/aspnetcore#47538]
I'm trying to create a Blazor WASM application that unifies setting a page's title (via `PageTitle` & containing a common "suffix") as well as showing the same title (but without the aforementioned suffix) within the main content area, too.

Since all those pages are rendered where I specify `@Body` within `MainLayout`, there's no chance to use `CascadingValue` or `Parameter`. Using `Templated components` as well as `Nested layouts` failed to solve the problem, too.

Currently, the only working solution involves calling `base.BuildRenderTree(__builder);` which, IMHO, isn't a good/valid approach as it's using an _internal_ Blazor field - which feels "hacky" right from the start:

`BasePage.razor`:
```csharp
<PageTitle>@Title — MySuffix</PageTitle>
@if (ShowTitle)
{
  <span>@Title</span>
}
@code
{
  protected virtual bool ShowTitle => true;
  protected virtual string Title => nameof(BasePage);
}
```

`Files.razor` as an example of a derived page:
```csharp
@page "/Files"
@inherits BasePage
@{
  base.BuildRenderTree(__builder); // This call shouldn't be required.
}
// HTML markup left out for brevity…
@code
{
  protected override string Title => nameof(Files);
}
```

Any help - either via enhancing Blazor or my code - would be greatly appreciated!

**Comment by Eagle3386** [comment]
Sorry for the late reply, been on vacation…

@MackinnonBuck: yes, there is - it breaks my "fallback scenario" which is this:

In reality, `BasePage.razor` is a bit more complex than my OP's code snippet outlined:
```razor
@inherits BlazorStateComponent
<PageTitle>@Title — MySuffix</PageTitle>
@if (ShowTitle)
{
  <span>@Title</span>
}
@code
{
  protected virtual bool ShowTitle => true;
  protected virtual string Title => Id.Remove(Id.IndexOf('-'));
}
```
As you can see, `BasePage.razor` itself inherits from [BlazorState](https://github.com/TimeWarpEngineering/blazor-state)'s `BlazorStateComponent` whose `Id` property is populated as `[ComponentName]-[counter, starting at 1]` upon run.

This represents kinda "automatic fallback scenario" as each page is created with its English name as the filename.

However, while writing this comment, I initially tried to explain the even greater benefit upon introduction of localization, i. e. switching to something like `Loc[Id.Remove(Id.IndexOf('-'))];` (as outlined [in the docs](https://learn.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?pivots=webassembly#localization)), in the future.
But instead of saving me from repetitive code for accessing the localized title, I ended up realizing, that I'd still have to overwrite `Title` - which in turn eliminates any "automatic usage" of BlazorState's `Id` property.

So, basically yes, your approach, i. e. turning `BasePage.razor` with `virtual` properties into a component with `Parameter` properties for usage in multiple places, is a much better one.
Thanks a lot for that - and your `@((RenderFragment)base.BuildRenderTree)` snippet, too! 👏🏻
